src/OpenTelemetrySDK/OpenTelemetryController.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Functions.PowerShellWorker.PowerShell;
+using Microsoft.Azure.Functions.PowerShellWorker.Utility;
+using LogLevel = Microsoft.Azure.WebJobs.Script.Grpc.Messages.RpcLog.Types.Level;
+
+namespace Microsoft.Azure.Functions.PowerShellWorker.OpenTelemetrySDK
+{
+    using PowerShell = System.Management.Automation.PowerShell;
+
+    internal class OpenTelemetryController
```

Comment:
Quite a lot of code without unit tests.

Reply:
Refactored the code to better work with the tests, and added the very first test. More is needed before this PR can be completed